examples/bri-3/src/bri/transactions/agents/transactions.agent.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { Transaction } from '../models/transaction';
+import { TransactionStatus } from '../models/transactionStatus.enum';
+
+import {
+  DELETE_WRONG_STATUS_ERR_MESSAGE,
+  NOT_FOUND_ERR_MESSAGE,
+  UPDATE_WRONG_STATUS_ERR_MESSAGE,
+} from '../api/err.messages';
+import { TransactionStorageAgent } from './transactionStorage.agent';
+import { BpiAccount } from '../../identity/bpiAccounts/models/bpiAccount';
+
+@Injectable()
+export class TransactionAgent {
+  constructor(private storageAgent: TransactionStorageAgent) {}
+  public throwIfCreateTransactionInputInvalid() {
+    return true;
+  }
+
+  public createNewTransaction(
+    id: string,
+    nonce: number,
+    workflowInstanceId: string,
+    workstepInstanceId: string,
+    from: BpiAccount,
+    to: BpiAccount,
+    payload: string,
+    signature: string,
+  ): Transaction {
+    return new Transaction(
+      id,
+      nonce,
+      workflowInstanceId,
+      workstepInstanceId,
+      from,
+      to,
+      payload,
+      signature,
+      TransactionStatus.Initialized,
+    );
+  }
+
+  public async fetchUpdateCandidateAndThrowIfUpdateValidationFails(
+    id: string,
+  ): Promise<Transaction> {
+    const transactionToUpdate = await this.storageAgent.getTransactionById(id);
+
+    if (!transactionToUpdate) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+
+    if (transactionToUpdate.status != TransactionStatus.Initialized) {
+      throw new BadRequestException(UPDATE_WRONG_STATUS_ERR_MESSAGE);
+    }
+
+    return transactionToUpdate;
+  }
+
+  public updateTransaction(
```

Comment:
@ognjenkurtic see my comment in #533 as to what elements in a transaction  could be updated

Reply:
resolved